### PR TITLE
【Hackathon No.93】增加 linux 下 cpu tensor file_descriptor 传输方案

### PR DESCRIPTION
**Pull request by zhangyuqin1998**

### CODE REVIEW

**Comment by ZHUI** [comment]
辛苦您提供的RFC方案，我下周review一下。感谢

File: rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+   ```
+
+# 六、测试和验收的考量
+
+测试考虑的case如下：
+
+1. 测试api是否可以正确执行
+2. 测试tensor是否可以被正确共享，数值计算是否正确
+3. 测试是否发生文件残留
+4. 输入Tensor的`dtype`为`float32`和`float64`时的结果正确性；
```
Comment (DesmonDay) [inline rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md:284]: 写单测时可增加多个dtype测试。

File: rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+本文档，主要设计完善`paddle.multiprocessing`模块。通过自定义Tensor序列化、反序列化方式，使用共享内存技术，实现paddle 的cpu Tensor在进程间快速传输、共享。
+
+## 2、功能目标
+
+完善 `paddle.multiprocessing`模块，可在多进程间，方便快捷的传输Tensor。
+
+## 3、意义
+
+ultiprocessing 是支持进程间 Tensor 传输的一种方式。
```
Comment (DesmonDay) [inline rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md:25]: m

File: rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+   
+   
+   def _rebuild_lodtensor_fd(cls, df, size, type_idx, dims, lod):
+       fd = df.detach()
+       lodtensor = cls._new_file_descriptor((fd, size, type_idx, dims, lod))
+       os.close(fd)
+       return lodtensor
+   ```
+
+3. 在paddle/fuild/memory/allocation/mmap_allocator.h和mmap_allocator.cc中修改MemoryMapAllocation并增加全局AllocateMemoryMapAllocationAndUnlink方法
```
Comment (ZHUI) [inline rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md:188]: 这块需要细化，跟torch的实现类似。走不通的 option file system file_desriptor 包括 后面的 mac win的存储申请，都走同一个函数。

Comment (ZHUI) [reply rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md:188]: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/MapAllocator.cpp#L66-L361

File: rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md (hunk 1 above)
Comment (zhangyuqin1998) [reply rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md:284]: > 写单测时可增加多个dtype测试。

好的

File: rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md (hunk 2 above)
Comment (zhangyuqin1998) [reply rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md:25]: > multiprocessing

好的

File: rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md (hunk 3 above)
Comment (zhangyuqin1998) [reply rfcs/APIs/20230222_design_for_cpu_tensor_file_descriptor.md]: > https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/MapAllocator.cpp#L66-L361

已修改

**Review by ZHUI — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
